### ISSUE TITLE
[Global] Contact and newsletter signup success state focus not well managed

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#777]
## Issue summary

When the contact or newsletter signup forms are submitted with valid data, a hash is added to the URL which causes focus to shift to the incorrect DOM node. Focus is meant to shift to the success heading on page load, however the heading does not receive keyboard focus. In some instances, this forces the user to navigate from the top of the document and not easily consume the success state. This would lead to a confusing user experience.

<details>
  <summary>Screenshots</summary>

  <img width="1310" alt="Dawn homepage. Highlighted sections include the contact form section, newsletter signup section, and newsletter signup which appears in the footer." src="https://user-images.githubusercontent.com/1392632/137516169-3c9241c7-aeab-45b9-8f99-05f2b03a890b.png">

</details>

#### HTML

```html
<!-- Contact form section… -->
<form
  method="post"
  action="/contact#ContactForm"
  id="ContactForm"
  accept-charset="UTF-8"
  class="contact-form"
>
  <!-- … -->
</form>

<!-- Newsletter signup form section… -->
<form
  method="post"
  action="/contact#contact_form"
  id="contact_form"
  accept-charset="UTF-8"
  class="newsletter-form"
>
  <!-- … -->
</form>

<!-- Newsletter signup form in footer -->
<form
  method="post"
  action="/contact#ContactFooter"
  id="ContactFooter"
  accept-charset="UTF-8"
  class="footer__newsletter newsletter-form"
>
  <!-- … -->
</form>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Include the Contact Form section and Newsletter section on the homepage, and Newsletter signup in the footer
3. With an active screen reader, fill in each form with valid data

### Expected

- Keyboard focus to shift to the success message heading element.

### Actual

- Focus shifts to the related `form` element which are not set to accept keyboard focus.

## Recommendation

Adjust each form `action` value by removing the hash portion of the URL. The success message headings allow for focus via `autofocus` and `tabindex` attributes. When the page loads, focus _should_ shift to the related heading.

#### HTML

```html
<!-- Contact form section… -->
<form
  method="post"
  action="/contact"
  id="ContactForm"
  accept-charset="UTF-8"
  class="contact-form"
>
  <!-- … -->
</form>

<!-- Newsletter signup form section… -->
<form
  method="post"
  action="/contact"
  id="contact_form"
  accept-charset="UTF-8"
  class="newsletter-form"
>
  <!-- … -->
</form>

<!-- Newsletter signup form in footer -->
<form
  method="post"
  action="/contact"
  id="ContactFooter"
  accept-charset="UTF-8"
  class="footer__newsletter newsletter-form"
>
  <!-- … -->
</form>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** Global
- **WCAG Principal:** Operable
- **WCAG SC:** [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html)
- **Severity:** High
- **Effort:** Low

**Comment by svinkle** [comment]
I figured it would be a platform issue. Who could we ping to get more eyes on this?
